manifests/rules/qemu.pp

```diff
@@ -41,7 +41,7 @@
   Boolean                                 $forward_traffic   = true,
   Boolean                                 $internal_traffic  = true,
   Boolean                                 $masquerade        = true,
-) {
+) inherits nftables {
```

Comment:
It's safe to assume that `nftables` will be present when including classes in `nftables::rules::*`. For the test to pass, just add:

```ruby
let(:pre_condition) { 'include nftables' }
```

before defining the contexts, something like:

```ruby
describe 'nftables::rules::qemu' do
  on_supported_os.each do |os, os_facts|
    context "on #{os}" do
      let(:facts) { os_facts }

      let(:pre_condition) { 'include nftables' }

      context 'default options' do
        it { is_expected.to compile }
...
```
https://rspec-puppet.com/documentation/classes/#specifying-code-to-include-before

This suggestion applies to more rules that you have patched, please review them.